libtransmission/crypto-utils.cc

```diff
@@ -106,15 +106,15 @@ int tr_rand_int_weak(int upper_bound)
 {
     TR_ASSERT(upper_bound > 0);
 
-    static bool init = false;
-
-    if (!init)
+    static std::mt19937 random_engine = []()
     {
-        srand(tr_time_msec());
-        init = true;
-    }
+        std::random_device random_device;
+        return std::mt19937{ random_device() };
+    }();
+    static std::uniform_int_distribution<> distribution{ 0 };
 
-    return rand() % upper_bound;
+    // Upper bound is inclusive in std::uniform_int_distribution.
+    return distribution(random_engine, std::uniform_int_distribution<>::param_type{ 0, upper_bound - 1 });
```

Comment:
Too wordy, my guess is this should work:
```suggestion
    return distribution(random_engine, { 0, upper_bound - 1 });
```

Reply:
Nope, constructor is explicit

Reply:
Hmm... Then maybe declare an alias for the distribution type to avoid duplicating it, or use `decltype(distribution)::param_type`?..

Reply:
Done!